rpc/wrtc_server_stream.go

```diff
@@ -277,6 +277,10 @@ func (s *webrtcServerStream) processHeaders(headers *webrtcpb.RequestHeaders) {
 		}
 	}
 
+	s.ch.server.peerConnsMu.RLock()
+	s.ch.server.processHeadersWorkers.Add(1)
+	s.ch.server.peerConnsMu.RUnlock()
```

Comment:
It's definitely possible my concurrent logic is not perfect here; would appreciate a couple pairs of eyes 😆 . The `RWMutex` + `WaitGroup` hurts my head, but I think it's probably the best way to synchronize this stuff.